core/trino-main/src/main/java/io/trino/security/ViewAccessControl.java

```diff
@@ -77,21 +73,15 @@ public void checkCanCreateViewWithSelectFromColumns(SecurityContext context, Qua
     }
 
     @Override
-    public void checkCanExecuteFunction(SecurityContext context, String functionName)
+    public boolean canExecuteFunction(SecurityContext context, QualifiedObjectName functionName)
     {
-        wrapAccessDeniedException(() -> delegate.checkCanGrantExecuteFunctionPrivilege(context, functionName, invoker, false));
+        return delegate.canCreateViewWithExecuteFunction(context, functionName);
```

Comment:
Hello @dain , is it intended to use `delegate.canCreateViewWithExecuteFunction()` instead of `delegate.canExecuteFunction()` here..? (I guess it is intended to use `VIEW DEFINED` permission by default)